src/components/NewsletterSignup/NewsletterSignup.mdx

```diff
@@ -0,0 +1,107 @@
+import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";
+
+import * as NewsletterSignupStories from "./NewsletterSignup.stories";
+import Link from "../Link/Link";
+
+<Meta of={NewsletterSignupStories} />
+
+# NewsletterSignup
+
+| Component Version | DS Version   |
+| ----------------- | ------------ |
+| Added             | `Prerelease` |
+| Latest            | `Prerelease` |
+
+## Table of Contents
+
+- {<Link href="#overview" target="_self">Overview</Link>}
+- {<Link href="#component-props" target="_self">Component Props</Link>}
+- {<Link href="#accessibility" target="_self">Accessibility</Link>}
+
+## Overview
+
+<Description of={NewsletterSignupStories} />
+
+## Component Props
+
+<Canvas of={NewsletterSignupStories.WithControls} />
+
+<ArgTypes of={NewsletterSignupStories.WithControls} />
+
+## Accessibility
+
+The `NewsletterSignup` component is a complex component built from various Reservoir
+DS and Chakra components.
+
+Within the `NewsletterSignup` component, the DS `form` component wraps around two DS `FormField` components.
+Those`FormField` components hold a DS `TextInput` component of `type="email"` and a DS `Button` component of `type="submit"` respectively.
+Each of these components has their own accessibility features documented in their respective Storybook pages.
```

Comment:
I found a great extension for VSCode called `Rewrap` that helps with wrapping and formatting long lines of text.

https://marketplace.visualstudio.com/items?itemName=stkb.rewrap